examples/console/agent.go

```diff
@@ -0,0 +1,370 @@
+	}
+	otherEscrowAccountSeqNum, err := getSeqNum(a.HorizonClient, otherEscrowAccountKey.Address())
+	if err != nil {
+		return fmt.Errorf("getting sequence number of other's escrow account: %w", err)
+	}
+	fmt.Fprintf(a.LogWriter, "escrow account seq: %v\n", escrowAccountSeqNum)
+	fmt.Fprintf(a.LogWriter, "other's escrow account seq: %v\n", otherEscrowAccountSeqNum)
+	a.channel = state.NewChannel(state.Config{
+		NetworkPassphrase: a.NetworkPassphrase,
+		MaxOpenExpiry:     openExpiry,
+		Initiator:         a.EscrowAccountKey.Address() > otherEscrowAccountKey.Address(),
+		LocalEscrowAccount: &state.EscrowAccount{
+			Address:        a.EscrowAccountKey,
+			SequenceNumber: escrowAccountSeqNum,
+		},
+		RemoteEscrowAccount: &state.EscrowAccount{
+			Address:        otherEscrowAccountKey,
+			SequenceNumber: otherEscrowAccountSeqNum,
+		},
+		LocalSigner:  a.EscrowAccountSigner,
+		RemoteSigner: otherSignerKey,
+	})
+	return nil
+}
+
+func (a *Agent) handleOpen(openIn state.OpenAgreement, send *json.Encoder) error {
+	open, authorized, err := a.channel.ConfirmOpen(openIn)
+	if err != nil {
+		return fmt.Errorf("confirming open: %w", err)
+	}
+	if authorized {
+		fmt.Fprintf(a.LogWriter, "open authorized\n")
+		if a.channel.IsInitiator() {
+			formationTx, err := a.channel.OpenTx()
+			if err != nil {
+				return fmt.Errorf("building formation tx: %w", err)
+			}
+			err = a.Submitter.SubmitFeeBumpTx(formationTx)
+			if err != nil {
+				return fmt.Errorf("submitting formation tx: %w", err)
+			}
+		}
+	}
+	if !open.Equal(openIn) {
+		err = send.Encode(message{Open: &open})
+		if err != nil {
+			return fmt.Errorf("encoding open to send back: %w", err)
+		}
+	}
+	return nil
+}
+
+func (a *Agent) handleUpdate(updateIn state.CloseAgreement, send *json.Encoder) error {
+	update, authorized, err := a.channel.ConfirmPayment(updateIn)
+	if errors.Is(err, state.ErrUnderfunded) {
+		fmt.Fprintf(a.LogWriter, "remote is underfunded for this payment based on cached account balances, checking their escrow account...\n")
+		var account horizon.Account
+		account, err = a.HorizonClient.AccountDetail(horizonclient.AccountRequest{AccountID: a.channel.RemoteEscrowAccount().Address.Address()})
+		if err != nil {
+			return fmt.Errorf("getting state of remote escrow account: %w", err)
+		}
+		var balance int64
+		balance, err = amount.ParseInt64(account.Balances[0].Balance)
+		if err != nil {
+			return fmt.Errorf("parsing balance of remote escrow account: %w", err)
+		}
+		fmt.Fprintf(a.LogWriter, "updating remote escrow balance to: %d\n", balance)
+		a.channel.UpdateRemoteEscrowAccountBalance(balance)
+		update, authorized, err = a.channel.ConfirmPayment(updateIn)
+	}
+	if err != nil {
+		return fmt.Errorf("confirming payment: %w", err)
+	}
+	if !update.Equal(updateIn) {
+		err = send.Encode(message{Update: &update})
+		if err != nil {
+			return fmt.Errorf("encoding payment to send back: %w", err)
+		}
+	}
+	if authorized {
```

Comment:
when sending a payment both participants get the `payment successfully received` log, even the participant that initially sent it

once again this might not be worth addressing now bc CAP-40 will change this code. So a `TODO` might make sense

Reply:
Fixed in 8a8965f.